### PR TITLE
Bump py-version and add python-typing-update to CI

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

We can tick off two things in #4683 😄

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -101,3 +101,16 @@ repos:
...
+      - id: python-typing-update
+        stages: [manual]
+        additional_dependencies:
+          - "black==22.3.0"
+        args:
+          - --py37-plus
+          - --force
+          - --keep-updates
+          - --black
+        files: ^(pylint)/.+\.py$
```
Comment (cdce8p) [inline .pre-commit-config.yaml:116]: I don't think we should add this. The existing pylint checks are much better.

File: .github/workflows/checks.yaml [hunk 2]
```diff
@@ -121,6 +121,7 @@ jobs:
           . venv/bin/activate
           pip install -e .
           pre-commit run pylint --all-files
+          pre-commit run --hook-stage manual python-typing-update --all-files
```
Comment (cdce8p) [inline .github/workflows/checks.yaml:124]: Same here

**Review by cdce8p — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Pierre-Sassoulas) [inline .pre-commit-config.yaml:116]: I agree with what Marc said earlier, that pylint's typing extension is vastly more efficient. I would rather add the typing extension to our pylintrc. I think we should only handle the bump of the py-version in this and discuss this in another one.

Comment (cdce8p) [reply .pre-commit-config.yaml:116]: > I would rather add the typing extension to our pylintrc.

It's already enabled. We just need to update `py-version`.

> I think we should only handle the bump of the py-version in this MR and discuss the pre-commit hook in another one.

AFAICT the typing changes here are from #6325.
With the typing extension, we don't need to add any other pre-commit hook 😄

Comment (DanielNoord) [inline .pre-commit-config.yaml:116]:
```suggestion
```

File: .github/workflows/checks.yaml (hunk 2 above)
Comment (DanielNoord) [inline .github/workflows/checks.yaml:124]:
```suggestion
```

**Comment by DanielNoord** [comment]
Changed to only the `py-version` change 👍

**Review by cdce8p — APPROVED** [review]

**Pull request merged**